### PR TITLE
Added additional resources & updated variables

### PR DESCRIPTION
**Pull request by robh007**
Just adding some updates

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -1,16 +1,53 @@
...
+
+resource "aws_dx_gateway_association" "this" {
+  count = "${var.create_dx_gateway ? 1 : 0}"
+
+  dx_gateway_id  =  "${aws_dx_gateway.this.*.id[0]}"
+  vpn_gateway_id = "${concat(element(aws_vpn_gateway.this.*.id, 0), list(var.vgw_id))}"
+}
+
+resource "aws_vpn_gateway" "this" {
+  count = "${var.create_dx_gateway && var.create_vgw ? 1 : 0}"
```
Comment (JoshiiSinfield) [inline main.tf:49]: @robh007 why is this tied to create_dx_gateway?

File: main.tf [hunk 2]
```diff
@@ -1,16 +1,53 @@
...
+
+  amazon_side_asn = "${var.bgp_asn}"
+  name            = "${var.gateway_name}"
+}
+
+resource "aws_dx_gateway_association" "this" {
+  count = "${var.create_dx_gateway ? 1 : 0}"
+
+  dx_gateway_id  =  "${aws_dx_gateway.this.*.id[0]}"
+  vpn_gateway_id = "${concat(element(aws_vpn_gateway.this.*.id, 0), list(var.vgw_id))}"
```
Comment (JoshiiSinfield) [inline main.tf:45]: same question as above

File: main.tf [hunk 3]
```diff
@@ -1,16 +1,53 @@
...
+  mtu              = "${var.mtu}"
 }
 
 resource "aws_dx_hosted_private_virtual_interface_accepter" "private_vif_accepter" {
+  count = "${var.create_dx_private_hosted_vif ? 1 : 0}"
+
   provider             = "aws.accepter"
   virtual_interface_id = "${aws_dx_hosted_private_virtual_interface.private_vif.id}"
-  vpn_gateway_id       = "${var.vgw_id}"
+  vpn_gateway_id       = "${concat(element(aws_vpn_gateway.this.*.id, 0), list(var.vgw_id))}"
```
Comment (JoshiiSinfield) [inline main.tf:19]: Does this work? vpn_gateway_id only takes a string, not a list. don't you need element(,0) on the outside of it all?

**Review by JoshiiSinfield — COMMENTED** [review]
Couple of questions

File: main.tf (hunk 2 above)
Comment (robh007) [reply main.tf:45]: You need a VGW to attach to the DX Gateway

File: main.tf (hunk 3 above)
Comment (robh007) [reply main.tf:19]: Fixed in latest commit.

**Pull request merged**